### PR TITLE
!!![TASK] Warn only once when a reference cannot be resolved

### PR DESCRIPTION
**Pull request by linawolf**
This is a breaking change to those who have implemented a custom ReferenceResolver.

### CODE REVIEW

File: packages/guides/src/ReferenceResolvers/DelegatingReferenceResolver.php [hunk 1]
```diff
@@ -20,21 +21,25 @@ public function __construct(private readonly iterable $resolvers, private readon
     {
     }
 
-    public function resolve(LinkInlineNode $node, RenderContext $renderContext): void
+    public function resolve(LinkInlineNode $node, RenderContext $renderContext, Messages|null $messages = null): void
     {
+        if ($messages === null) {
+            $messages = new Messages();
+        }
```
Comment (jaapio) [inline packages/guides/src/ReferenceResolvers/DelegatingReferenceResolver.php:28]:
```suggestion
        $messages ??= new Messages();
```

**Review by jaapio — COMMENTED** [review]
Why are the messages nullable? was this an attempt to not make this a breaking change?

File: packages/guides/src/ReferenceResolvers/DelegatingReferenceResolver.php (hunk 1 above)
Comment (linawolf) [reply packages/guides/src/ReferenceResolvers/DelegatingReferenceResolver.php:28]: the DelegatingReferenceResolver is calling all the other resolvers and takes care of the warnings. However it  implements the same Interface. We could however move the message handling to ReferenceResolverPreRender...

Comment (linawolf) [reply packages/guides/src/ReferenceResolvers/DelegatingReferenceResolver.php]: I moved the Message handling to the PreRenderer. Please have another look

**Review by jaapio — APPROVED** [review]

**Pull request merged**